docker/start.py

```diff
@@ -205,36 +205,43 @@ def main(args, environ):
             config_dir, config_path, environ, ownership
         )
 
-    if mode is not None:
-        error("Unknown execution mode '%s'" % (mode,))
+    args = ["python"] + args[1:]
 
-    config_dir = environ.get("SYNAPSE_CONFIG_DIR", "/data")
-    config_path = environ.get("SYNAPSE_CONFIG_PATH", config_dir + "/homeserver.yaml")
+    if "-m" not in args:
+        args = ["-m", synapse_worker] + args
 
-    if not os.path.exists(config_path):
-        if "SYNAPSE_SERVER_NAME" in environ:
-            error(
-                """\
-Config file '%s' does not exist.
+    if not filter(lambda p: p.startswith("--config-path"), args):
```

Comment:
I think `any` is clearer here:

```suggestion
    if not any(p.startswith("--config-path") for p in args):
```